### PR TITLE
[coordinator] Docker integration tests for prom-remote storage

### PR DESCRIPTION
**Pull request by Antanukas**
**What this PR does / why we need it**:

In this PR docker integration test suite is implemented to cover Prom Remote Write backend use case from https://github.com/m3db/m3/pull/3742.

**Special notes for your reviewer**:

For tests we setup:

1. etcd instance
2. coordinator admin
2. coordinator with Prom Remote Write backend
3. 2 aggregators
4. 2 prometheus instances that accept remote writes `--enable-feature=remote-write-receiver`

Coordinator has 2 endpoints setup `prometheusraw` and `prometheusagg`. Metrics are written to coordinator using remote write API `http://m3coordinator01:7201/api/v1/prom/remote/write`. There is a rollup rule that gets applied and metric gets renamed before sending to aggregated prometheus.

Both prometheus are queried and are expected to return a metric.

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
None
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
None
```

### CODE REVIEW

File: scripts/docker-integration-tests/prom_remote_write_backend/utils.sh [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  network=$(docker network ls | grep -F $network_name | tr -s ' ' | cut -f 1 -d ' ' | tail -n 1)
+
+  out=$( (docker run -it --rm --network "$network"          \
+    "$PROMREMOTECLI_IMAGE"                                  \
+    -u http://m3coordinator01:7201/api/v1/prom/remote/write \
+    -t __name__:"${metric_name}"                            \
+    -d "${datapoint_timestamp}","${datapoint_value}" | grep -v promremotecli_log) || true)
+
+  success=$(echo "$out" | grep -v promremotecli_log | docker run --rm -i "$JQ_IMAGE" jq .success)
+  status=$(echo "$out" | grep -v promremotecli_log | docker run --rm -i "$JQ_IMAGE" jq .statusCode)
```
Comment (vpranckaitis) [inline scripts/docker-integration-tests/prom_remote_write_backend/utils.sh:26]: Why are these being piped though jq docker container? Could it pipe through local jq instead (i.e.`... | jq ...`) like in other functions in this file?

**Review by vpranckaitis — APPROVED** [review]

File: scripts/docker-integration-tests/prom_remote_write_backend/utils.sh (hunk 1 above)
Comment (Antanukas) [reply scripts/docker-integration-tests/prom_remote_write_backend/utils.sh:26]: I am not sure. My guess - to not force installation on JQ on a running machine. I copied (and removed some uneeded stuff) from other tests.

Comment (vpranckaitis) [reply scripts/docker-integration-tests/prom_remote_write_backend/utils.sh:26]: That seems reasonable. However, IIUC, other code in the same file uses local jq, e.g.

https://github.com/m3db/m3/blob/36e3845d11718947810455e60fb0a39f0f071317/scripts/docker-integration-tests/prom_remote_write_backend/utils.sh#L50-L51

So we still have to have jq on machine, with the additional complexity of running some of the jq commands through docker container.

Comment (Antanukas) [reply scripts/docker-integration-tests/prom_remote_write_backend/utils.sh]: ah yeah, actually I used local myself in couple of commands without thinking. Will use local here as well.

**Pull request merged**